src/App.tsx

```diff
@@ -1,61 +1,129 @@
+
+    const newTodo = {
+      id: maxId + 1,
+      title: titleInput,
+      completed: false,
+      userId: selectedUser.id,
+      user: selectedUser,
+    };
+
+    setTodos(prevTodos => ([
+      ...prevTodos,
+      newTodo,
+    ]));
+
+    resetForm();
+  };
+
+  const handleSubmitTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setFormChecked(true);
+
+    if (titleInput && userSelectedByName && titleInput.trim() !== '') {
+      addNewTodo();
+    }
+  };
+
+  const handleChangeTitle = (
+    (event: React.FormEvent<HTMLInputElement>) => (
+      setUserSelectedByName(event.currentTarget.value)));
+
+  const handleChangeUser = (
+    (event: React.FormEvent<HTMLSelectElement>) => (
+      setUserSelectedByName(event.currentTarget.value)));
 
-export const App = () => {
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/users" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+    <div>
+      <div className="App">
+        <h1>Add todo form</h1>
+
+        <form
+          action="/api/users"
+          method="POST"
+          onSubmit={handleSubmitTodo}
+        >
+
+          <div className="field" />
+          <label>
+            {'Title: '}
+            <input
+              type="text"
+              value={titleInput}
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              onChange={handleChangeTitle}
+            />
+          </label>
+
+          {(isFormChecked && !titleInput) && (
```

Comment:
you can create a variable to store this condition 
for example
`const isEmptyInput = isFormChecked && !titleInput;
`
```suggestion
          {isEmptyInput  && (
```